services/tests/startup.rs

```diff
@@ -0,0 +1,139 @@
+use assert_cmd::prelude::*;
+
+use geoengine_services::test_data;
+use serial_test::serial;
+use std::{
+    io::BufRead,
+    process::{Command, Stdio},
+};
+
+#[cfg(feature = "pro")]
+#[serial]
+#[tokio::test(flavor = "multi_thread", worker_threads = 1)]
+async fn it_starts_postgres_backend_with_no_warnings() {
+    use geoengine_services::util::config::get_config_element;
+
+    const SCHEMA_NAME: &str = "it_starts_postgres_backend_with_no_warnings";
+
+    async fn run_server_and_check_warnings() {
+        let mut server = Command::cargo_bin("main")
+            .unwrap()
+            .env("GEOENGINE_WEB__BACKEND", "postgres")
+            .env("GEOENGINE_POSTGRES__SCHEMA", SCHEMA_NAME)
+            .stderr(Stdio::piped())
+            .spawn()
+            .unwrap();
+
+        let stderr = server.stderr.take().expect("failed to read stderr");
+
+        let mut reader = std::io::BufReader::new(stderr);
+
+        let mut startup_succesful = false;
+        for _ in 0..99 {
+            let mut buf = String::new();
+            reader.read_line(&mut buf).unwrap();
+
+            // println!("{buf}");
+
+            assert!(!buf.contains("WARN"));
+
+            if buf.contains("Tokio runtime found") {
+                startup_succesful = true;
+                break;
+            }
+        }
+
+        server.kill().unwrap();
+
+        assert!(startup_succesful);
+    }
+
+    // change cwd s.t. the config file can be found
+    std::env::set_current_dir(test_data!("..")).expect("failed to set current directory");
+
+    // create a fresh schema for this test
+    let config = get_config_element::<geoengine_services::util::config::Postgres>().unwrap();
+    let (client, connection) = tokio_postgres::connect(
+        &format!(
+            "host={host} port={port} user={user} password={password} dbname={database}",
+            host = config.host,
+            user = config.user,
+            port = config.port,
+            password = config.password,
+            database = config.database
+        ),
+        tokio_postgres::NoTls,
+    )
+    .await
+    .unwrap();
+
+    // The connection object performs the actual communication with the database,
+    // so spawn it off to run on its own.
+    tokio::spawn(async move {
+        if let Err(e) = connection.await {
+            eprintln!("connection error: {e}");
+        }
+    });
```

Comment:
Unfortunately, I still don't get why you are doing this.

Reply:
this is from the example on how to connect to postgres https://docs.rs/tokio-postgres/latest/tokio_postgres/

Reply:
Ok, but you don't need to wait for it at the end?

Reply:
if I don't await the connection, the test never completes, so unfortunately it seems we need this code. 

awaiting it at the end doesn't work.

Reply:
No, I meant awaiting the join handle of the spawn, not the inner await of the connection.

Reply:
if I await it, the test never stops, because the connection never closes.

I could do

```
drop(client);
conn_handle.await.unwrap();
```

then it stops correctly. But I don't really see the point in doing this.